llmfoundry/models/hf/hf_causal_lm.py

```diff
@@ -157,6 +158,24 @@ def __init__(self, om_model_config: Union[DictConfig,
             if dist.get_local_rank() != 0 and init_device == 'mixed':
                 om_model_config.pretrained = False
 
+            # If the HuggingFace model is coming from a local folder, Hugging Face copies the modules into the
+            # transformers modules cache. On particular systems, this operation seems to cause contention between
+            # the different processes. To avoid this contention, we first create the model (on meta device) on local rank
```

Comment:
It would work, but it would be _much_ slower because for rank 0 it would have to init on cpu.

Reply:
ah yeah, that's right. i was thinking for meta initialization the total time is about the same: if we initialize on cpu first, the total time for initializing all ranks would be cpu + meta. If we initialize on meta first, the total time would be meta + max(cpu, meta) == meta + cpu. 

But then for cpu initialization it would be twice as slow.